creating_packages/toolchains/cmake.rst

```diff
@@ -0,0 +1,170 @@
+
+.. code:: python
+
+    def toolchain(self):
+        tc = CMakeToolchain(self)
+        tc.definitions["MYVAR"] = "MyValue"
+        tc.definitions.debug["MYCONFIGVAR"] = "MyDebugValue"
+        tc.definitions.release["MYCONFIGVAR"] = "MyReleaseValue"
+        return tc
+
+This will be translated to:
+
+- One ``set()`` definition for ``MYVAR`` in ``conan_toolchain.cmake`` file.
+- One ``set()`` definition, using a cmake generator expression in ``conan_project_include.cmake`` file,
+  using the different values for different configurations. It is important to recall that things
+  that depend on the build type cannot be directly set in the toolchain.
+
+generators
+----------
+
+The ``CMakeToolchain`` only works with the ``cmake_find_package`` and ``cmake_find_package_multi``
+generators. Using others will raise, as they can have overlapping definitions that can conflict.
+
+
+CMake build helper
+------------------
+
+The ``CMake()`` build helper that works with the ``CMakeToolchain`` is also experimental,
+and subject to breaking change in the future. It will evolve to adapt and complement the
+toolchain functionality. It supports the following methods:
+
+constructor
++++++++++++
+
+.. code:: python
+
+    def __init__(self, conanfile, generator=None, build_folder=None, parallel=True,
+                 msbuild_verbosity="minimal"):
+
+- ``conanfile``: the current recipe object. Always use ``self``.
+- ``generator``: CMake generator. Define it only to override the default one (like ``Visual Studio 15``).
+  Note that as the platform (x64, Win32...) is now defined in the toolchain it is not necessary to specify it here.
+- ``build_folder``: Relative path to a folder to contain the temporary build files
+- ``parallel``: Set it to ``False`` to deactivate using parallel builds. If activated, it will use
+  ``cpu_count`` configuration as the number of parallel jobs to use.
+- ``msbuild_verbosity``: Used to define the output of MSBuild builds.
+
+
+configure()
++++++++++++
+
+.. code:: python
+
+    def configure(self, source_folder=None):
+
+Calls ``cmake``, with the given generator and passing ``-DCMAKE_TOOLCHAIN_FILE=conan_toolchain.cmake``.
+It will also provide the CMake generator in the command like, like ``-G "Visual Studio 15"``. Note
+that it is not necessary to specify the platform, like ``-G "Visual Studio 15 Win64"``, as the
+platform is already defined in the toolchain file.
+
+- ``source_folder``: Relative path to the folder containing the root *CMakeLists.txt*
+
+
+build()
++++++++
+
+.. code:: python
+
+    def build(self, build_type=None, target=None):
+
+
+Calls the build system. Equivalent to :command:`cmake --build .` in the build folder.
+
+
+- ``build_type``: Use it only to override the value defined in the ``settings.build_type``. It 
+  can fail if the build is single configuration (e.g. Unix Makefiles), as in that case the build
+  type must be specified at configure time, not build type.
```

Comment:
I'd make this a little bit more clear:

_"Use it only to override the value defined in the ``settings.build_type`` for a multi-configuration generator (e.g. Visual Studio, XCode). This value will be ignored for single-configuration generators, they will use the one defined in the toolchain file during the install step"_